### PR TITLE
test(toggle): start implementing toggle tests

### PR DESCRIPTION
**Pull request by redox**
I also moved from the Template to templateProps everywhere

### CODE REVIEW

**Comment by redox** [comment]
It's still WIP, but I changed too much things around the `Template`->`templateProps` so I would merge that one first, and keep working once merged.

File: widgets/toggle/toggle.js [hunk 1]
```diff
@@ -68,11 +59,7 @@ function toggle({
         transformData: prepareToggleData(transformData),
         defaultTemplates,
         templatesConfig,
-        templates: {
-          header: templates.header,
-          item: templates.body,
```
Comment (redox) [inline widgets/toggle/toggle.js]: I restored `item` over `body`; I found it very confusing; no?

Comment (vvo) [inline widgets/toggle/toggle.js]: Yes indeed, I tried to be smart. This will be a BREAKING CHANGE (only for history/search, because nobody uses this :))

Comment (redox) [inline widgets/toggle/toggle.js]: done

**Comment by vvo** [comment]
L
G
T
M

**Pull request merged**